master/lustrefactory.py

```diff
@@ -6,26 +6,32 @@
 from buildbot.steps.shell import ShellCommand
 from buildbot.steps.shell import Configure
 from buildbot.steps.shell import SetPropertyFromCommand 
-from buildbot.steps.transfer import FileUpload
+from buildbot.steps.transfer import FileUpload, FileDownload
 from buildbot.steps.trigger import Trigger
 from buildbot.status.results import SUCCESS 
 from buildbot.status.results import FAILURE 
 from buildbot.status.results import SKIPPED 
 from buildbot.status.results import WARNINGS
 
-def do_step_build(step, name):
+def do_step_if_value(step, name, value):
     props = step.build.getProperties()
-    if props.hasProperty(name) and props[name] == "yes":
+    if props.hasProperty(name) and props[name] == value:
         return True
     else:
         return False
 
+def do_step_build(step, name):
+    return do_step_if_value(step, name, "yes")
```

Comment:
As much as I hate doing that as well, its required. The doStepIf argument of a build step expects a function where the only parameter is the step object itself. So those have to stay.